src/test/llm/llmnode_test.cpp

```diff
@@ -1284,6 +1284,72 @@ TEST_P(LLMFlowHttpTestParameterized, unaryChatCompletionsStopStringExceedingSize
         ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
 }
 
+TEST_P(LLMFlowHttpTestParameterized, unaryChatCompletionsPromptExceedsModelMaxLength) {
+    auto params = GetParam();
+    if (params.modelName.find("vlm") != std::string::npos) {
+        GTEST_SKIP();
+    }
+    std::string prompt;
+    for (int i = 0; i < 1024; i++) {
+        prompt += "prompt";
+    }
+    std::string requestBody = R"(
+        {
+            "model": ")" + params.modelName +
+                              R"(",
+            "stream": false,
+            "seed" : 1,
+            "messages": [
+            {
+                "role": "user",
+                "content": ")" +
+                              prompt + R"("
+            }
+            ]
+        }
+    )";
+
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointChatCompletions, requestBody, &response, comp, responseComponents, writer),
+        ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+}
+
+TEST_P(LLMFlowHttpTestParameterized, unaryChatCompletionsStoppedByModelMaxLength) {
+    auto params = GetParam();
+    if (params.modelName.find("vlm") != std::string::npos) {
+        GTEST_SKIP();
+    }
+    std::string prompt;
+    for (int i = 0; i < 1023; i++) {
+        prompt += "prompt";
+    }
+    std::string requestBody = R"(
```

Comment:
Is model length 2048 here? 
In such case maybe it would be worth to check if max_tokens=1000 and ignore_eos=True would work?